internal/server/SimLoop.go

```diff
@@ -0,0 +1,148 @@
+package server
+
+import (
+	"SOMAS2023/internal/common/objects"
+	"SOMAS2023/internal/common/utils"
+	"SOMAS2023/internal/common/voting"
+	"fmt"
+	"sort"
+
+	"math"
+	"slices"
+
+	"github.com/google/uuid"
+)
+
+func (s *Server) RunSimLoop(iterations int) {
+
+	s.ResetGameState()
+	s.FoundingInstitutions()
+
+	// run this for n iterations
+	for i := 0; i < iterations; i++ {
+		s.RunRoundLoop()
+	}
+
+}
+
+func (s *Server) ResetGameState() {
+	// kick everyone off bikes
+	for id, _ := range s.GetAgentMap() {
+		if bikeId, ok := s.megaBikeRiders[id]; ok {
+			s.megaBikes[bikeId].RemoveAgent(id)
+			delete(s.megaBikeRiders, id)
+		}
+	}
+
+	// respawn people who died in previous round (conditional)
+	if utils.RespawnAtRound {
+		for _, agent := range s.deadAgents {
+			s.AddAgent(agent)
```

Comment:
We probably want something that updates the agent's energy level once it's respawned, otherwise it'll just die again next round